plugin/core/registry.py

```diff
@@ -38,18 +38,46 @@ def get_position(view: sublime.View, event: Optional[dict] = None, point: Option
         return None
 
 
+class LspWindowCommand(sublime_plugin.WindowCommand):
+    """
+    Inherit from this class to define requests which are not bound to a particular view. This allows to run requests
+    for example from links in HtmlSheets or when an unrelated file has focus.
+    """
+
+    # When this is defined in a derived class, the command is enabled only if there exists a session with the given
+    # capability attached to a view in the window.
+    capability = ''
+
+    # When this is defined in a derived class, the command is enabled only if there exists a session with the given
+    # name attached to a view in the window.
+    session_name = ''
+
+    def is_enabled(self) -> bool:
+        return self.session() is not None
+
+    def session(self) -> Optional[Session]:
+        for session in windows.lookup(self.window)._sessions:
```

Comment:
Would prefer adding a getter for sessions rather than accessing private property.